pkg/agent/flannel/flannel.go

```diff
@@ -103,30 +103,36 @@ func LookupExtInterface(iface *net.Interface, netMode int) (*backend.ExternalInt
 	}
 	logrus.Debugf("The interface %s will be used by flannel", iface.Name)
 
-	ifaceAddr, err = ip.GetInterfaceIP4Addr(iface)
-	if err != nil {
-		return nil, fmt.Errorf("failed to find IPv4 address for interface %s", iface.Name)
+	if netMode == (ipv4+ipv6) || netMode == ipv4 {
+		ifaceAddr, err = ip.GetInterfaceIP4Addrs(iface)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find IPv4 address for interface %s", iface.Name)
+		}
+		logrus.Infof("The interface %s with ipv4 address %s will be used by flannel", iface.Name, ifaceAddr[0])
+	} else {
+		ifaceAddr = append(ifaceAddr, nil)
 	}
-	logrus.Infof("The interface %s with ipv4 address %s will be used by flannel", iface.Name, ifaceAddr)
 
 	if netMode == (ipv4 + ipv6) {
-		ifacev6Addr, err = ip.GetInterfaceIP6Addr(iface)
+		ifacev6Addr, err = ip.GetInterfaceIP6Addrs(iface)
 		if err != nil {
 			return nil, fmt.Errorf("failed to find IPv6 address for interface %s", iface.Name)
 		}
 
-		logrus.Infof("Using dual-stack mode. The ipv6 address %s will be used by flannel", ifacev6Addr)
+		logrus.Infof("Using dual-stack mode. The ipv6 address %s will be used by flannel", ifacev6Addr[0])
+	} else {
+		ifacev6Addr = append(ifacev6Addr, nil)
```

Comment:
What about using a switch + case statement:
```
switch netMode:
case ipv4:
    ifaceAddr, err = ip.GetInterfaceIP4Addrs(iface)
    logs
case (ipv4+ipv6):
    ifaceAddr, err = ip.GetInterfaceIP4Addrs(iface)
    logs
    ifacev6Addr, err = ip.GetInterfaceIP6Addrs(iface)
    logs
case ipv6:
    ifacev6Addr, err = ip.GetInterfaceIP6Addrs(iface)
    logs
```
I think it might be easier to follow